src/hide.rs

```diff
@@ -63,7 +63,7 @@ mod nightly {
 }
 
 // When a C compiler is available, a dummy C function can be used.
-#[cfg(not(feature = "no_cc"))]
+#[cfg(all(not(feature = "no_cc"), feature = "std"))]
 mod cc {
     use std::os::raw::c_void;
```

Comment:
Wow, using `std::os::raw::c_void` from `no_std` is one deep rabbit hole. Some decent background in this thread:

https://internals.rust-lang.org/t/solve-std-os-raw-c-void/3268/16

I guess this is the path forward, but work has not yet begun:

https://github.com/rust-lang/rust/issues/43467